packages/vite/src/node/server/openBrowser.ts

```diff
@@ -38,6 +39,15 @@ export function openBrowser(
   return false
 }
 
+export function resolveBrowserUrl(server: ViteDevServer): string {
+  const options = server.config.server
+  const hostname = resolveHostname(options.host)
+  const port = options.port || 3000
+  const protocol = options.https ? 'https' : 'http'
+  const path = typeof options.open === 'string' ? options.open : options.base
+  return `${protocol}://${hostname.name}:${port}${path ?? '/'}`
+}
+
```

Comment:
do we need to make it a common function to export hostname、port and others
@Shinigami92  WDYT